### ISSUE TITLE
Make use of reified type parameters for View DSL to allow factories installation

### ISSUE THREAD

**Issue by LouisCAD** [LouisCAD/Splitties#103]
Here's some inspiration: https://helw.net/2018/08/06/appcompat-view-inflation/

Here's how it could look: `v<TextView>(…) { … }`, and it'd check the type and delegate instantiation to installed factories. There could also be a way to scope View types delegates (not the same as Kotlin property delegates).

**Comment by LouisCAD** [comment]
The `textView(ctx: Context)` and similar methods currently in the `viewdsl-appcompat-styles` split could be deprecated, before their name is reused to be an alias of `v<TextView>`, that would be agnostic to whether AppCompat and/or Google Material Design library are used.

**Comment by LouisCAD** [comment]
Part of the prototype:
```kotlin
import android.content.Context
import android.support.annotation.IdRes
import android.support.annotation.StyleRes
import android.view.View
import splitties.viewdsl.core.NO_THEME
import splitties.viewdsl.core.Ui
import splitties.viewdsl.core.wrapCtxIfNeeded

inline fun <reified V : View> Context.v2(
        @IdRes id: Int = View.NO_ID,
        @StyleRes theme: Int = NO_THEME,
        initView: V.() -> Unit = {}
): V = V::class.java.getConstructor(Context::class.java).newInstance(wrapCtxIfNeeded(theme)).also {
    it.id = id
}.apply(initView)

inline fun <reified V : View> View.v2(
        @IdRes id: Int = View.NO_ID,
        @StyleRes theme: Int = NO_THEME,
        initView: V.() -> Unit = {}
) = context.v2(id, theme, initView)

inline fun <reified V : View> Ui.v2(
        @IdRes id: Int = android.view.View.NO_ID,
        @StyleRes theme: Int = NO_THEME,
        initView: V.() -> Unit = {}
) = ctx.v2(id, theme, initView)
```

**Comment by LouisCAD** [comment]
Maybe add a second optional parameter for style. This could be used for different button styles. Maybe it could also be used with interface types?

How would it work with interface types?
What could it be with interfaces instead?
And with inline classes?
And what about multiplatform support? Could abstracting Views and ViewGroups away allow sharing more code?

**Comment by LouisCAD** [comment]
The view factories could be put in the `Context` as a system service so it can then be accessed from any `View` `Context` with proper scope.

In fact, that's how the `LayoutInflater` is retrieved.

**Comment by LouisCAD** [comment]
### Any help on the API design is welcome

**Comment by LouisCAD** [comment]
Released in 2.0.0-alpha6 (and also alpha5)
